### PR TITLE
Product-Module

### PR DESCRIPTION
**Pull request by sonikavirani**
In Product-module API done the CRUD operation and 
Working on Store and get the Product-image.

### CODE REVIEW

File: src/actions/productAction.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const productData = {
+      productName,
+      price,
+      userId,
+      ProductImage,
+      categoryId,
+      quantity
+    };
+    productModel.insert(productData).then((data) => {
+      res.json({status: true, data, message: 'Product Added'})
```
Comment (suresh-jarvis) [inline src/actions/productAction.js]: Wrong message string :  it should be `Product added successfully.`

File: src/actions/productAction.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }).catch(() => {
+      res.json({status: false, message: 'Product Not Added'})
```
Comment (suresh-jarvis) [inline src/actions/productAction.js]: Change message here also

File: src/model/products.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      Products.find({categoryId}, function (error, data) {
+        resolve((error) ? [] : (data == null) ? [] : data)
+      })
+    })
+  },
+
+  delete: function (id) {
+    return new Promise((resolve) => {
+      Products.findByIdAndRemove(id, function (error, data) {
+        resolve((error) ? {} : (data == null) ? {} : data)
```
Comment (suresh-jarvis) [inline src/model/products.js]: Delete query will always return in Boolean value

File: src/model/products.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    return new Promise((resolve, reject) => {
+      this.findByProductId(id).then((resData) => {
+        if (Object.keys(resData).length > 0) {
+          Object.keys(data).map(rKey => {
+            resData[rKey] = data[rKey];
+          });
+          resData['updatedAt'] = moment();
+
+          resData.save(function (error, data) {
+            error == null ? resolve(data) : reject(false);
```
Comment (suresh-jarvis) [inline src/model/products.js]: Use only Resolve dont use Reject

File: src/routes/productRoute.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express.Router();
+const productAction = require('../../../product-mgmt-api/src/actions/productAction');
```
Comment (suresh-jarvis) [inline src/routes/productRoute.js]: Change this path it's not correct way

Comment (suresh-jarvis) [reply src/routes/productRoute.js]: require('../actions/productAction');

**Comment by sonikavirani** [comment]
Error Handling on CRUD operation

File: src/model/products.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    return new Promise((resolve) => {
+      this.findByProductId(id).then((resData) => {
+        if (Object.keys(resData).length > 0) {
+          Object.keys(data).map(rKey => {
+            resData[rKey] = data[rKey];
+          });
+          resData['updatedAt'] = moment();
+
+          resData.save(function (error, data) {
+            error ? resolve(false) : resolve(data)
```
Comment (suresh-jarvis) [inline src/model/products.js]: Use only single resolver using ternary operator logic : Like this : `resolve((err) ? false : data)`

**Comment by sonikavirani** [comment]
Resolve all comments,
Check duplicate Product name and working on Product Image.

**Pull request merged**